src/analyses/static_analysis.h

```diff
@@ -86,6 +86,9 @@ class domain_baset
   // this computes the join between "this" and "b"
   // return true if "this" has changed
 
+  bool has_been_seen() const { return seen; }
+  void set_seen() { seen=true; }
+
 protected:
   bool seen;
```

Comment:
This data member should be private, to ensure clients only modify it through the new public interface.